src/Chapter4.hs

```diff
@@ -472,10 +477,12 @@ Implement the Applicative instance for our 'Secret' data type from before.
 -}
 instance Applicative (Secret e) where
     pure :: a -> Secret e a
-    pure = error "pure Secret: Not implemented!"
+    pure = Reward
 
     (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
-    (<*>) = error "(<*>) Secret: Not implemented!"
+    (<*>) (Trap e) _ = Trap e
+    (<*>) _ (Trap e) = Trap e
+    (<*>) (Reward f) (Reward a) = Reward (f a)
```

Comment:
This correct! 💯 
You can also notice that these two cases look very similar to the Functor instance of `Secret` 🗺️ 
You can actually reuse `fmap` 🙂

Reply:
Oh, nice!